Containers/imaginary/Dockerfile

```diff
@@ -1,29 +1,21 @@
-# From https://github.com/h2non/imaginary/blob/master/Dockerfile
-FROM nextcloud/imaginary:20230301
+FROM golang:1.20.3-alpine3.17 as go
+RUN apk add --no-cache \
+        vips-dev \
+        build-base; \
+    go install github.com/h2non/imaginary@master # use master branch as base

```

Comment:
I am not sure if this is enough. Woudln't all the dependencies like libpng and else also need to be added to the base before compiling imaginary? Also I would really like to have some kind of versioning here so that it would not suddenly break if there should be a new commit upstream.

Reply:
vips-dev comes with a lot of deps and libs, I've compared the schared libaries using the ldd command and this install method has all libaries
to versioning, since they don't tag version, we could use commit shas and create a github workflow to update the sha

Reply:
I see, 

> vips-dev comes with a lot of deps and libs, I've compared the schared libaries using the ldd command and this install method has all libaries

I see, So e.g. pdf previews would still work? I guess best would be to compare with these mimetypes: https://github.com/nextcloud/server/blob/efe2fb66c78a4b9a7865d5f1d969434d1d465473/lib/private/Preview/Imaginary.php#L60

> to versioning, since they don't tag version, we could use commit shas and create a github workflow to update the sha

Yes, that sounds good!

Reply:
I've compared the shared libaries and it seems that alpine has even more shared libaries and more updated libaries

Reply:
the workflow is missing

Reply:
`git ls-remote https://github.com/h2non/imaginary master | cut -f1` that is the command to get the latest commit sha from the master branch

Reply:
should I create a new workflow file or modify an existing one?

Reply:
I guess it would probably make sense to create a new one.

Reply:
could you please do it? I have for the next week only a mobile phone with me, no laptop/desktop

Reply:
I've added one, not sure if it works, since I've created it on a mobile device

Reply:
if it works, it should create a PR with the long commit sha

Reply:
see below :)